warp_journal/server.py

```diff
@@ -206,13 +206,15 @@ def _update_warp_history(self):
             if not url:
                 url = url_util.find_gacha_url()
             region, auth_token = url_util.extract_region_and_auth_token(url)
+            api_endpoint = url_util.extract_api_endpoint(url)
         except (AuthTokenExtractionError, LogNotFoundError) as e:
             bottle.response.status = 400
             logging.warning('Unable to extract auth token: %s', e)
             return {'message': str(e)}
 
         logging.debug('Extracted region and token: %s; %s', region, auth_token)
         self._client.set_region_and_auth_token(region, auth_token)
+        self._client.set_api_endpoint(api_endpoint)
```

Comment:
You're correct to adapt this style, but I didn't like this way of handling data statefully, so I refactored it.